pdns/lazy_allocator.hh

```diff
@@ -35,12 +35,12 @@ struct lazy_allocator {
                    "lazy_allocator must only be used with trivial types");
 
     pointer
-    allocate (size_type const n) {
+    allocate (size_type const& n) {
```

Comment:
I'm not sure passing a `size_t` by const reference is a good idea, is `cppcheck` really issuing a warning about that?